packages/medusa/src/api/store/shipping-options/[id]/calculate/route.ts

```diff
@@ -1,19 +1,34 @@
 import { HttpTypes } from "@medusajs/framework/types"
 import { MedusaRequest, MedusaResponse } from "@medusajs/framework/http"
 import { calculateShippingOptionsPricesWorkflow } from "@medusajs/core-flows"
+import { ContainerRegistrationKeys } from "@medusajs/framework/utils"
 
 export const POST = async (
   req: MedusaRequest<HttpTypes.StoreCalculateShippingOptionPrice>,
-  res: MedusaResponse<HttpTypes.StoreCalculateShippingOptionPriceResponse>
+  res: MedusaResponse<HttpTypes.StoreShippingOptionResponse>
 ) => {
+  const query = req.scope.resolve(ContainerRegistrationKeys.QUERY)
+
   const { result } = await calculateShippingOptionsPricesWorkflow(
     req.scope
   ).run({
     input: {
-      shipping_option_ids: [req.params.id],
+      shipping_options: [{ id: req.params.id, data: req.validatedBody.data }],
       cart_id: req.validatedBody.cart_id,
     },
   })
 
-  res.status(200).json(result[0])
+  const { data } = await query.graph({
+    entity: "shipping_option",
+    fields: req.remoteQueryConfig.fields,
+    filters: { id: req.params.id },
+  })
+
+  const shippingOption = data[0]
+
+  shippingOption.calculated_price = result[0]?.calculated_amount
+  shippingOption.is_calculated_price_tax_inclusive =
+    result[0]?.is_calculated_price_tax_inclusive
```

Comment:
super nit(non-blocking): We could probably make this change at the fulfillment (provider) interface level to avoid this reshaping – but we can punt that for later

Reply:
you mean rename `calculated_amount` to `calculated_price` and use spread?